src/Helper/Log/TelemetryClient.php

```diff
@@ -44,26 +67,141 @@ public function __construct(Amplitude $amplitude, bool $telemetry_is_enabled, ?s
    *   TRUE if telemetry is ready or FALSE if not.
    */
   public function isReady(): bool {
-    return (bool) $this->amplitude;
+    return (bool) $this->domoClient;
+  }
+
+  /**
+   * @param $data
+   * Dataset in JSON format
+   *
+   * @return object|array The result with the dataset ID, or an error.
+   */
+  public function createDataSet(array $data) {
```

Comment:
Per the Single Responsibility Principle, this class should have only one reason to change. As the Domo client, that reason should be due to API differences with Domo. Generating the data to send should belong to a class with only _that_ responsibility. See `\Acquia\Orca\Helper\Log\TelemetryEventPropertiesBuilder`, for example. In fact, you should modify that class according to your needs, renaming it to `OrcaEventPropertiesBuilder` if the logic is Domo-specific, and use it.